gcpkms/keysource.go

```diff
@@ -233,14 +234,32 @@ func (key *MasterKey) newKMSClient() (*kms.KeyManagementClient, error) {
 	return client, nil
 }
 
-// getGoogleCredentials returns the SopsGoogleCredentialsEnv variable, as
+// getGoogleCredentials returns the connection options taken from the
+// SopsGoogleCredentialsEnv or SopsGoogleCredentialsOAuthEnv variables, as
 // either the file contents of the path of a credentials file, or as value in
-// JSON format. It returns an error if the file cannot be read, and may return
-// a nil byte slice if no value is set.
-func getGoogleCredentials() ([]byte, error) {
-	defaultCredentials := os.Getenv(SopsGoogleCredentialsEnv)
-	if _, err := os.Stat(defaultCredentials); err == nil {
-		return os.ReadFile(defaultCredentials)
-	}
-	return []byte(defaultCredentials), nil
+// JSON format. It returns an error if the file cannot be read, and will return
+// a nil value if neither environment variables are set.
+func getGoogleCredentials() (option.ClientOption, error) {
+	if envCredentials, isSet := os.LookupEnv(SopsGoogleCredentialsEnv); isSet {
+		credentials := []byte(envCredentials)
+		if _, err := os.Stat(envCredentials); err == nil {
+			if credentials, err = os.ReadFile(envCredentials); err != nil {
+				return nil, err
```

Comment:
Perhaps don't exit here, but instead check `SopsGoogleCredentialsOAuthEnv` as well?

Reply:
@warwick-mitchell1 I believe this is also done by the application default credentials: https://cloud.google.com/docs/authentication/application-default-credentials#search_order

Reply:
You maybe right, this isn't the part I'm really adding with this PR though, this was already like this. I'm adding the Oauth token below which isn't covered by application-defaultcreds

Reply:
I understand this. However, with your added functionality I believe it makes sense to also adjust the old behavior. If any fail condition happens with the credentials by file lookup, continue to check whether credentials by token area available before failing completely.